tools/walletextension/wallet_extension.go

```diff
@@ -238,14 +289,23 @@ func (we *WalletExtension) handleSubmitViewingKey(resp http.ResponseWriter, req
 		return
 	}
 
-	// We have to drop the leading "0x", and transform the V from 27/28 to 0/1.
+	// We drop the leading "0x", and transform the V from 27/28 to 0/1.
 	signature, err := hex.DecodeString(reqJSONMap["signature"][2:])
 	if err != nil {
 		logAndSendErr(resp, fmt.Sprintf("could not decode signature from client to hex: %s", err))
 		return
 	}
 	signature[64] -= 27
 
+	// We recover the public key address.
+	msgToSign := ViewingKeySignedMsgPrefix + hex.EncodeToString(we.viewingPublicKeyBytes)
+	recoveredPublicKey, err := crypto.SigToPub(accounts.TextHash([]byte(msgToSign)), signature)
+	if err != nil {
+		logAndSendErr(resp, fmt.Sprintf("could not recover public key from signature: %s", err))
+		return
+	}
+	we.viewingPublicKeyAddress = crypto.PubkeyToAddress(*recoveredPublicKey)
```

Comment:
the default "from" is basically set to the metamsk key that was last used to sign a viewing key.

This should be added in the comment where you define the: viewingPublicKeyAddress

Basically that it is the default "from" and where it comes from